### DISCUSSION TITLE
Connect relations with a multi field id

### DISCUSSION THREAD

**Discussion by yuedai0531** [prisma/prisma#4956] [Q&A] [answered]
Hi, I currently have these models in my schema. There's a many Messages to One Conversation relation.
```prisma
model Conversation {
  createdAt DateTime  @default(now())
  lawyer    Lawyer    @relation(fields: [lawyerId], references: [id])
  lawyerId  String
  human     Human     @relation(fields: [humanId], references: [id])
  humanId   String
  messages  Message[]
  @@id([humanId, lawyerId])
}

model Message {
  id                    String       @default(uuid()) @id
  conversation          Conversation @relation(fields: [conversationHumanId, conversationLawyerId], references: [humanId, lawyerId])
  conversationHumanId   String
  conversationLawyerId  String
  conversationId        String
  content               String
  createdAt             DateTime     @default(now())
  senderUserType        UserType
}

```

My question is how do I connect Messages to **Conversation that has a multi field id** with the prisma client

```ts
  async createMessage(
    message: CreateMessageDto
  ): Promise<Message> {
    return await this.prismaService.message.create({
      data: {
        conversation: {
          connect: {
            id: message.conversationId
          }
        },
        content: message.content,
        senderUserType: message.senderUserType
      }
    })
  }
```
This is what I've tried and the error is 
`TS2322: Type '{ id: string; }' is not assignable to type 'ConversationWhereUniqueInput'.   Object literal may only specify known properties, and 'id' does not exist in type 'ConversationWhereUniqueInput'.`

I have also tried 
```
        conversation: {
          connect: {
            humanId_lawyerId: message.conversationId
          }
        },
```
and 
```
        conversation: {
          connect: {
            humanId: message.humanId,
            lawyerId: message.lawyerId
          }
        },
```
Thanks!

**Comment by ryands17** [comment, marked as answer]
Hey @yuedai0531 👋 

In this case, you would need to do the following:

```ts
await prisma.message.create({
    data: {
      content: 'some-content',
      conversation: {
        connect: {
          humanId_lawyerId: {
            humanId: 'human-id',
            lawyerId: 'lawyer-id',
          },
        },
      },
    },
})
```

If you press <kbd>ctrl+space</kbd> inside the `connect` block in VSCode, you will the required suggestion :)

**Reply by yuedai0531** [reply]
thank you!!
I'm on Webstorm, what extension are you using for autocomplete?

**Reply by nikolasburk** [reply]
Hey @yuedai0531, I'm not a Webstorm user but the autocompletion should just be powered by TypeScript, so I don't think there should be an extension needed for it 🤔 do you generally not get autocompletion when using TypeScript, or is the issue only with Prisma Client API calls?

That being said, there is a Prisma plugin for Jetbtrains IDEs that e.g. helps with syntax highlighting and the Prisma schema: https://plugins.jetbrains.com/plugin/14240-prisma But I don't think this should affect the TypeScript portion of your work...

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
